commit/plugin.go

```diff
@@ -159,6 +160,21 @@ func NewPlugin(
 		reporter,
 	)
 
+	var reportBuilder reportBuilderFunc
+
+	// These options were added to allow for more flexibility around report building. For example Solana
+	// only supports a single merkle root per report.
+	maxRoots := offchainCfg.MaxMerkleRootsPerReport
+	if !offchainCfg.RMNEnabled && maxRoots != 0 {
+		if offchainCfg.MultipleReportsEnabled {
+			reportBuilder = buildMultipleReports
+		} else {
+			reportBuilder = buildTruncatedReport
```

Comment:
So this is enabled if:
* RMNEnabled == false
* MaxMerkleRootsPerReport > 0
* MultipleReportsEnabled == false
So we want to have potentially multiple roots per report, but only a single report (based on the code), isn't that just a standard report?

Or do we want to special case this on MaxMerkleRootsPerReport == 1? Cuz if you set that to 2, you'd get multiple roots per report.

Reply:
>  So we want to have potentially multiple roots per report, but only a single report (based on the code), isn't that just a standard report?

It's not a standard report, this lets you have a single truncated report. Without this feature using `MaxMerkleRootsPerReport` without `MultipleReportsEnabled` will be a configuration error.